### ISSUE TITLE
Feature Request: Upsamling1D

### ISSUE THREAD

**Issue by ht93** [pytorch/pytorch#1501]
Hi, I am new to PyTorch and trying to immigrate my code to PyTorch. I found there is no Upsampling1D in PyTorch which is one of key layer for my network.
Hope you can add Upsampling1D to PyTorch.
Thanks in advance!

**Comment by apaszke** [comment]
What kind of upsampling do you need specifically?

**Comment by ht93** [comment]
@apaszke Thanks for the reply!
I was using `Upsampling1D` with same mechanism of `UpsamplingNearest2d`. I was using Keras before, their `Upsampling1D` only has this option.
So I need the `Upsampling1D` with same mechanism of `UpsamplingNearest2d`. (if `Bilinear` type is available in the future, I think it would also be helpful.)

Thank you very much!

**Comment by ncullen93** [comment]
Hey while we are at it... I am looking for an 3D upsampling! Could probably implement it in pure pytorch but not sure that's a good long term solution. I see `nn.MaxUnpool3d` but I think that's something different entirely.

**Comment by soumith** [comment]
there's a pr for 3d up sampling that i need to review and merge

**Comment by apaszke** [comment]
@ht93 if it's a nearest mode then fortunately there's an easy workaround for now - use `unsqueeze` and pretend that your input is 2D (with e.g. height of 1), use `UpsamplingNearest2d`, and then slice off a single row (all rows should be the same).

**Comment by ht93** [comment]
@apaszke Thanks for the advise.
I found two ways to implement this from the tutorial, not sure which one is better since I am not very cleaer about whether the backward and gradient updating would work fine with this two implementation. 
Shall I use it like http://pytorch.org/tutorials/beginner/pytorch_with_examples.html#pytorch-custom-nn-modules :
```
class Upsampling1D(torch.nn.Module):
    def __init__(self, scale_factor):
        super(Upsampling1D, self).__init__()
        self.scale_factor = scale_factor
        self.upsampling2D = nn.UpsamplingNearest2d(scale_factor=scale_factor)

    def forward(self, x):
        x = torch.unsqueeze(x,3)
        x = self.upsampling2D(x)
        x = x[:,:,:,0]
        return x
```
or implement it inside my Net() class like http://pytorch.org/tutorials/beginner/blitz/neural_networks_tutorial.html#define-the-network: 
```
class Net(nn.Module)
    #some __init__ and forward function here

    def unsampling1D(self, x, scale_factor):
        x = torch.unsqueeze(x,3)
        upsampling2D = nn.UpsamplingNearest2d(scale_factor=scale_factor)
        x = upsampling2D(x)
        x = x[:,:,:,0]
```

**Comment by apaszke** [comment]
I think either are fine, but the first one seems cleaner as you're nicely encapsulating the logic in a single place unrelated to your network.

**Comment by soumith** [comment]
fixed via https://github.com/pytorch/pytorch/pull/2846
